packages/lockfile-lint-api/src/validators/ValidateHost.js

```diff
@@ -38,7 +39,10 @@ module.exports = class ValidateHost {
           return REGISTRY[hostValue] ? REGISTRY[hostValue] : hostValue
         })
 
-        if (!allowedHosts.includes(packageResolvedURL.host)) {
+        const isPassing =
+          isValidHost(packageMetadata.resolved, allowedHosts) ||
+          isAllowedUrl(packageMetadata.resolved, (options && options.allowedUrls) || [])
+        if (!isPassing) {
```

Comment:
Not exactly what I meant originally :-)
You added `ValidateUrl.js` as a class for the lockfile-lint-api - perfect 👌
Now, instead of hard coding the logic of how to resolve hosts and URLs into the api package like you are doing here, you should actually do that same logic but in the lockfile-lint CLI package. So this needs to be taken out, and as a result, the tests you added to `packages/lockfile-lint-api/__tests__/validators.host.test.js ` should actually be done on the lockfile-lint CLI package.

Does that make sense?

Reply:
@lirantal so, something like this then (I haven't actually tested this code yet but wondering if this is what you are thinking about):

```javascript
function ValidateHostManager ({path, type, validatorValues, validatorOptions}) {
  debug('validate-host-manager')(
    `invoked with validator options: ${JSON.stringify(validatorValues)}`
  )

  const options = {
    lockfilePath: path,
    lockfileType: type
  }

  const parser = new ParseLockfile(options)
  const lockfile = parser.parseSync()
  const validator = new ValidateHost({packages: lockfile.object})
  const validationResult = validator.validate(validatorValues, validatorOptions)

  // Check if some of the errors are for allowed URLs and filter those out
  if (validatorOptions && validatorOptions.allowedUrls) {
    validationResult.errors = validationResult.errors.filter(result => {
      const packageUrl = lockfile.object[result.package].resolved

      return !isAllowedUrl(packageUrl, validatorOptions.allowedUrls)
    })

    // If we don't have any errors left at this point make sure it's a success type
    if (!validationResult.errors.length) {
      validationResult.type = 'success'
    }
  }

  return validationResult
}
```

Reply:
Yes this is the direction I was intending to go with.
I see you are referencing `isAllowedUrl()` which is supposed to be related to `ValidateUrl`, which I guess you're doing because it's not possible to just call `validator.validate` the same way because that goes on the entire package list, and here we need to do it one by one? If so, maybe it's worth to just add a method of like `validateSingle()` to that `ValidateUrl` class?

Overall, seems ok. Let's proceed.

Reply:
@lirantal ok, adding `validateSingle()` method makes sense to me.

Reply:
@lirantal alright, I went with this approach. Let me know what adjustments you would like to see.

Reply:
Taking a look 👀